datasets/census_bureau_acs/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,550 @@
+    for key in group_id:
+        logging.info(f"reading data from API for KPI {key}...")
+        # replece = {
+        #     "+year_report+": year_report,
+        #     "+key[0:-3]+": key[0:-3],
+        #     "+key[-3:]+": key[-3:],
+        #     "+api_naming_convention+": api_naming_convention,
+        # }
+        # source_url_new = string_replace(source_url, replece)
+        str1 = source_url.replace("+year_report+", year_report)
+        str2 = str1.replace("+key[0:-3]+", key[0:-3])
+        str3 = str2.replace("+key[-3:]+", key[-3:])
+        source_url_new = str3.replace("+api_naming_convention+", api_naming_convention)
+        try:
+            r = requests.get(source_url_new, stream=True)
+            logging.info(f"Source url : {source_url_new}")
+            logging.info(f"status code : {r.status_code}")
+            if r.status_code == 200:
+                text = r.json()
+                frame = pd.DataFrame(text)
+                frame = frame.iloc[1:, :]
+                frame["KPI_Name"] = key
+                list_temp.append(frame)
+        except OSError as e:
+            logging.info(f"error : {e}")
+            pass
+    logging.info("creating the dataframe...")
+    df = pd.concat(list_temp)
+    return df
+
+
+def extract_data_and_convert_to_df_state_level(
+    group_id: dict,
+    state_code: dict,
+    year_report: str,
+    api_naming_convention: str,
+    source_url: str,
+) -> pd.DataFrame:
+    list_temp = []
+    for key in group_id:
+        for sc in state_code:
+            logging.info(f"reading data from API for KPI {key}...")
+            logging.info(f"reading data from API for KPI {sc}...")
+            # replece = {
+            #     "+year_report+": y ear_report,
+            #     "+key[0:-3]+": key[0:-3],
+            #     "+key[-3:]+": key[-3:],
+            #     "+api_naming_convention+": api_naming_convention,
+            #     "+sc+": sc,
+            # }
+            str1 = source_url.replace("+year_report+", year_report)
+            str2 = str1.replace("+key[0:-3]+", key[0:-3])
+            str3 = str2.replace("+key[-3:]+", key[-3:])
+            str4 = str3.replace("+api_naming_convention+", api_naming_convention)
+            source_url_new = str4.replace("+sc+", sc)
+            # source_url = string_replace(source_url, replece)
+            try:
+                r = requests.get(source_url_new, stream=True)
+                logging.info(f"Source url : {source_url_new}")
+                logging.info(f"status code : {r.status_code}")
+                if r.status_code == 200:
+                    text = r.json()
+                    frame = pd.DataFrame(text)
+                    frame = frame.iloc[1:, :]
+                    frame["KPI_Name"] = key
+                    list_temp.append(frame)
+            except OSError as e:
+                logging.info(f"error : {e}")
+            except TypeError as error:
+                logging.inf(error)
+                time.sleep(5)
```

Comment:
Let's remove this. Or the whole except block

Reply:
removed